### PR TITLE
added a function to return resource address from source and seeds

### PR DESCRIPTION
**Pull request by dhruvja**
### Description
This PR is related to the issue [#3739](https://github.com/aptos-labs/aptos-core/issues/3739) created to add a function in TS SDK to fetch the resource account given that the source address and seeds are provided.
### Test Plan
I have added a function getResourceAccountAddress with 2 parameters sourceAddress and seeds which are used to derive the resource account. Since the resource account derivation is expensive on chain, we can use this method to derive if off chain and send it to the smart contract.
This is second PR i am opening, the last PR had some issues with package.json and yarn.lock. The old PR [#3822](https://github.com/aptos-labs/aptos-core/pull/3822) was closed. A test is also added to check the functionality.

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/4918)

### CODE REVIEW

File: ecosystem/typescript/sdk/src/aptos_account.test.ts [hunk 1]
```diff
@@ -65,3 +66,11 @@ test("Signs Strings", () => {
     "0xc5de9e40ac00b371cd83b1c197fa5b665b7449b33cd3cdd305bb78222e06a671a49625ab9aea8a039d4bb70e275768084d62b094bc1b31964f2357b7c1af7e0d",
   );
 });
+
+test("Gets the resource account address", () => {
+  const sourceAddress = "9db1d65a321e4a86f8098cb04e76fce098890b84211fa06d65f24dc644bf0fec";
+  const seedInHex = "3031" // hexadecimal form of "01" which is 3031
+  const seed = new HexString(seedInHex).toUint8Array();
+
+  expect(AptosAccount.getResourceAccountAddress(sourceAddress, seed).hex()).toBe("0x915e47f986471a5faba9cb2f726611cd554344419ddd6d5b3e9f00bcafd30169");
```
Comment (blockchainlover2019) [inline ecosystem/typescript/sdk/src/aptos_account.test.ts:75]: Hey, how did you get this address in contract side?
**"0x915e47f986471a5faba9cb2f726611cd554344419ddd6d5b3e9f00bcafd30169"**

I followed your method, but I can't get proper resource account address.

Comment (dhruvja) [reply ecosystem/typescript/sdk/src/aptos_account.test.ts:75]: i used those seeds and address and printed the resource address created by the module using the `create_resource_address` function.

Comment (blockchainlover2019) [reply ecosystem/typescript/sdk/src/aptos_account.test.ts:75]: I found the solution.
```
  const originBytes = new Uint8Array(source.length + seed.length + 1);
  originBytes.set(source);
  originBytes.set(seed, source.length);
  originBytes.set([255], source.length + seed.length);
```
You omitted to add 255 at the end of bytes.
Please look at here.
https://github.com/aptos-labs/aptos-core/blob/83641d3f083762f31700060e5e95fbb1cbc28bf0/aptos-move/framework/aptos-framework/sources/account.move#L400

**Comment by davidiw** [comment]
This needs to be updated and auto formatted / linted. Please update, closing out cause it is a bit stale.

**Pull request closed without merging**